pkg/sysregistriesv2/system_registries_v2_test.go

```diff
@@ -650,101 +650,68 @@ func TestPullSourcesFromReference(t *testing.T) {
 		SystemRegistriesConfDirPath: "testdata/this-does-not-exist",
 	}
 	registries, err := GetRegistries(sys)
-	assert.Nil(t, err)
-	assert.Equal(t, 2, len(registries))
-
-	// Registry A allowing any kind of pull from mirrors
-	registryA, err := FindRegistry(sys, "registry-a.com/foo/image:latest")
-	assert.Nil(t, err)
-	assert.NotNil(t, registryA)
-	// Digest
-	referenceADigest := toNamedRef(t, "registry-a.com/foo/image@sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa")
-	pullSources, err := registryA.PullSourcesFromReference(referenceADigest)
-	assert.Nil(t, err)
-	assert.Equal(t, 3, len(pullSources))
-	assert.Equal(t, "mirror-1.registry-a.com/image@sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", pullSources[0].Reference.String())
-	assert.True(t, pullSources[1].Endpoint.Insecure)
```

Comment:
The tests for.Reference.String() and .Insecure` have not been fully migrated:

The test data should probably add an optional `tagInsecure, digestInsecure []bool` arrays that contain the expected values. Similarly for `Reference`, I guess the arrays could contain the repo, so that the test appends `digest` or `tag` appropriately.

Reply:
(I don’t think it’s necessary to add this data to to existing table entries dealing with digest/tag mirroring: if the arrays are `nil` in the test data, that part of the comparison can be skipped.)

Reply:
added the test data